src/app/core/tasks/marathon/mailserver/model.nim

```diff
@@ -214,18 +209,24 @@ proc cycleMailservers(self: MailserverModel) =
   if self.activeMailserver != "":
     info "Disconnecting active mailserver", peer=self.activeMailserver
     self.nodes[self.activeMailserver] = MailserverStatus.Disconnected
-    if self.wakuVersion == 2:
-      dropPeerByID(self.activeMailserver)
+    if self.wakuVersion == WAKU_VERSION_2:
+      self.mailserverController.dropPeerByID(self.activeMailserver)
     else:
-      removePeer(self.activeMailserver)
+      self.mailserverController.removePeer(self.activeMailserver)
     self.activeMailserver = ""
   self.findNewMailserver()
 
 proc checkConnection*(self: MailserverModel) {.async.} =
   while true:
     info "Verifying mailserver connection state..."
-    let pinnedMailserver = status_settings.getPinnedMailserver()
-    if self.wakuVersion == 1 and pinnedMailserver != "" and self.activeMailserver != pinnedMailserver:
+    var pinnedMailserver = ""
+    if(not self.mailserverController.isNil):
+      pinnedMailserver = self.mailserverController.getPinnedMailserver()
+
+    echo "\n\n\n\nPINNED MS: ", repr(pinnedMailserver)
+    echo "WAKU V: ", self.wakuVersion
+    echo "activeMailserver: ", self.activeMailserver
```

Comment:
Are those echos still needed?

Reply:
@jrainville this is needed more than anything else. ;) Removed, thanks.